src/main/java/uk/gov/register/service/ItemValidator.java

```diff
@@ -64,6 +66,17 @@ private void validateFieldsValue(JsonNode inputEntry, Map<String, Field> fields)
         });
     }
 
+    private void dataTypeBasedValidation(final JsonNode inputEntry, final String fieldName, final Datatype datatype,
+                                         final JsonNode fieldValue) {
+        if (DataType.DATETIME.getName().equals(datatype.getName())) {
+            if (!DateFormatChecker.isDateFormatValid(fieldValue.textValue())) {
```

Comment:
We could pass this conditional check to the existing `throwEntryValidationExceptionIfConditionIsFalse` method, which a) removes a layer of nesting `if` statements here and b) means that we don't an additional `throwEntryValidationException` method